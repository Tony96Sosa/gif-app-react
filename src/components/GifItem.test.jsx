import { render, screen } from "@testing-library/react";
import { GifItem } from "./GifItem";

describe("Prueba en <GifItem/>", () => {
  const title = "Goku";
  const url = "https://dragon-ball.com/Goku.jpg";

  test("Deberia matchear con su snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar el alt y src indicados", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    const { alt, src } = screen.getByRole("img");
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });

  test("Debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
