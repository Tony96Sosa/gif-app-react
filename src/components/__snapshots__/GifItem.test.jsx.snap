// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifItem/> Deberia matchear con su snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://dragon-ball.com/Goku.jpg"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
